File: packages/python/scripts/copy-data.py
#!/usr/bin/env python3
"""Copies the shared data/ files from the repo root into the Python
package source tree so they're included in the wheel + sdist.

PEP 517 builds run in an isolated /tmp/ directory which can't see
relative paths like ../../data/. Copying the JSON files into the
package source dir solves it for both `pip install -e .` and `python
-m build`.

Path resolution in detector.py prefers the installed-package location
and falls back to the source-checkout location, so running tests
against the source tree without this copy still works.

Wired to the publish workflow as a pre-build step; also runs from a
plain `python -m build` if you copy first.
"""
import shutil
from pathlib import Path

HERE = Path(__file__).resolve().parent
REPO_ROOT_DATA = HERE.parent.parent.parent / "data"
DEST = HERE.parent / "src" / "disposable_email_detector" / "data"

if DEST.exists():
    shutil.rmtree(DEST)
DEST.mkdir(parents=True)

for name in [
    "disposable-domains.json",
    "forwarding-alias-domains.json",
    "personally-observed-domains.json",
]:
    src = REPO_ROOT_DATA / name
    shutil.copy2(src, DEST / name)
    print(f"copied {name}")

File: packages/js/scripts/copy-data.mjs
// Copies the shared data/ files from the repo root into packages/js/data/
// so they're included in the npm tarball. Wired to `prepublishOnly` in
// package.json — runs automatically before `npm publish`.
//
// Path resolution in src/domains.js prefers the local copy (published
// layout) and falls back to ../../../data (source-checkout layout), so
// running tests in the dev tree doesn't require this copy to exist.

import { copyFileSync, mkdirSync, existsSync, rmSync } from "node:fs";
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";

const here = dirname(fileURLToPath(import.meta.url));
const repoRootData = join(here, "..", "..", "..", "data");
const dest = join(here, "..", "data");

if (existsSync(dest)) rmSync(dest, { recursive: true });
mkdirSync(dest, { recursive: true });

const files = [
  "disposable-domains.json",
  "forwarding-alias-domains.json",
  "personally-observed-domains.json",
];
for (const f of files) {
  copyFileSync(join(repoRootData, f), join(dest, f));
  console.log(`copied ${f}`);
}
